framework/include/base/FEProblem.h

```diff
@@ -1271,5 +1265,36 @@ FEProblem::finalizeUserObjects(const MooseObjectWarehouse<T> & warehouse)
   }
 }
 
+template <class T>
+T &
+FEProblem::getGlobal()
+{
+  // get the class name as map key
+  const std::string name = typeid(T).name();
```

Comment:
I know about demangle, but this is uses the type name as a map key only and does not report it to the user. All I need here is for `typeid(T).name()` to return a consistent result for successive calls with the same `T`.